design-proposals/multiarch-cluster.md

```diff
@@ -0,0 +1,154 @@
+## Introduction
+
+A more indepth proposal can be found at the following google doc https://docs.google.com/document/d/1KYc3lsHHhri0gbCZTOGe5lz2dNJogRRwrXwN9Do9kT0/edit?usp=sharing
+
+### Background & Recent Discussions
+As it stands, Kubevirt has provided support for both X86 and ARM-based clusters but the assumption is that the architecture of the cluster is homogeneous. 
+
+This means that all components are compiled for the architecture where they will be running. The codebase uses the runtime package to determine the compiled architecture for applicable logic branches in the code. 
+
+For users running a mixture of VM workloads, the only currently available option is to introduce multiple clusters. The complexities associated with federating multiple clusters in unison can serve as a deterrent to the usefulness of KubeVirt in these types of use cases.
+
+### Purpose 
+This document aims to explain the justification for any changes required to KubeVirt in order to provide the ability for users to run in a multi-architecture environment.  In addition, this document aims to provide an open forum to discuss whether the proposed changes are the most effective way of providing the desired functionality.
+
+## Motivation
+
+The motivation for adding the proposed functionality stemmed from a personal use case that required running X86 and ARM-based workloads. After investigation, the conclusion reached was that no such functionality was currently available in KubeVirt. This is largely due to the lack of users working with ARM-based workloads. 
+
+Regardless of this fact, it was identified that the framework already has accomplished most of the heavy lifting by providing support for each of the architecture exclusively. Given this, the amount of effort required to expand the existing logic was identified to be minimal and a POC was created. 
+
+## Goals
+
+- Support the deployment of Kubevirt into multi-architecture clusters
+- Ability to specify architecture when deploying VM/VMIs 
+- Feature changes should not break any existing functionality in KubeVirt
+- Multi-architecture support should be placed behind a feature-gate and be an optional config item based on user-need
+
+## User Stories
+
+As a user, I want to be able to create and start one or more VM/VMIs which allow me to explicitly specify the architecture of the workload
```

Comment:
As a user, I want to be able to run an ARM compute-plane with an x86 control-plane and vice versa.